compiler/hie-core/src/Development/IDE/LSP/Notifications.hs

```diff
@@ -0,0 +1,100 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE RankNTypes #-}
+
+module Development.IDE.LSP.Notifications
+    ( addNotifications
+    ) where
+
+import           Development.IDE.LSP.Protocol
+import           Development.IDE.LSP.Server hiding (runServer)
+import qualified Language.Haskell.LSP.Core as LSP
+
+import Development.IDE.Types.Logger
+import Development.IDE.Core.Service
+import Development.IDE.Types.Location
+
+import qualified Data.Set                                  as S
+import qualified Data.Text as T
+
+import Development.IDE.Core.FileStore
+import Development.IDE.Core.OfInterest
+
+import qualified Network.URI                               as URI
+
+
+textShow :: Show a => a -> T.Text
+textShow = T.pack . show
+
+
+addNotifications :: RunHandler -> LSP.Handlers -> IO LSP.Handlers
```

Comment:
Should that be called `addNotificationHandlers`? `addNotifications` sounds like we are emitting notifications which we are not.